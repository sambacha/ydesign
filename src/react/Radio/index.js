import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { themeGet } from '@styled-system/theme-get'
import { RadioButtonChecked } from '@rimble/icons/es/md'
import { RadioButtonUnchecked } from '@rimble/icons/es/md'
import theme from '../theme'
import Box from '../Box'
import Text from '../Text'

const StyledWrapper = styled(Box)`
  & {
    position: relative;
    display: inline-block;
  }

  > input {
    cursor: pointer;
    appearance: none;
    position: relative;
    z-index: 1;
    display: block;
    height: 1.5rem;
    width: 1.5rem;
    margin: 0;
  }

  > svg {
    position: absolute;
    z-index: 0;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    height: 100%;
    width: 100%;
    pointer-events: none;
    fill: ${(props) => themeGet('colors.grey', '#CCC')};
  }

  > svg[name='checked'] {
    display: none;
    z-index: 1;
    fill: ${(props) => themeGet('colors.primary', '#000')};
  }

  > input:checked ~ svg[name='checked'] {
    display: block;
  }
`

const StyledLabel = styled(Box)`
  & {
    cursor: pointer;
  }
  &:hover input:not(:disabled) ~ svg[name='unchecked'] {
    fill: currentColor;
  }
`

StyledWrapper.defaultProps = {
  theme,
}

const Radio = React.forwardRef(
  ({ className, label, name, value, id, ...props }, ref) => (
    <StyledLabel
      forwardedAs={'label'}
      display={'flex'}
      alignItems={'center'}
      className={className}
      {...props}
      htmlFor={id}
      opacity={props.disabled ? 0.4 : 1}
    >
      <StyledWrapper theme={props.theme}>
        <input
          type={'radio'}
          name={name}
          value={value}
          id={id}
          ref={ref}
          {...props}
        />
        <RadioButtonChecked name={'checked'} />
        <RadioButtonUnchecked name={'unchecked'} />
      </StyledWrapper>
      <Text
        color={'text'}
        fontSize={1}
        fontWeight={3}
        lineHeight={1}
        ml={1}
        mr={2}
      >
        {label}
      </Text>
    </StyledLabel>
  )
)

Radio.defaultProps = {
  theme,
  color: 'primary',
}

Radio.propTypes = {
  ...Box.propTypes,
  theme: PropTypes.object,
  label: PropTypes.string,
}

Radio.displayName = 'Radio'

export default Radio
