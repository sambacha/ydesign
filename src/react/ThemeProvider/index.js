import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import theme from '../theme';

const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  );
};

ThemeProvider.propTypes = {
  theme: PropTypes.object,
};

ThemeProvider.defaultProps = {
  theme,
};

ThemeProvider.displayName = 'ThemeProvider';

export default ThemeProvider;
