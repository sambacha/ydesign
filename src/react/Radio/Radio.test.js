import Radio from './';
import ThemeProvider from '../ThemeProvider';
import theme from '../theme';

describe('Radio component sanity', () => {
  it('has name', () => {
    expect(Radio.displayName).toBe('Radio');
  });

  it('matches default snapshot', () => {
    const component = render(<Radio />);
    expect(component).toMatchSnapshot();
  });

  it('matches themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Radio />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
