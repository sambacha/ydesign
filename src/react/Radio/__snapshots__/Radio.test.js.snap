// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio component sanity matches default snapshot 1`] = `
.c4 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: inherit;
}

.c5 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #4E3FCE;
  opacity: 1;
}

.c2 {
  box-sizing: border-box;
}

.c6 {
  box-sizing: border-box;
  color: #3F3D4B;
  font-size: 14px;
  font-weight: 600;
  line-height: 1;
  margin-left: 4px;
  margin-right: 8px;
}

.c3 {
  position: relative;
  display: inline-block;
}

.c3 > input {
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
  z-index: 1;
  display: block;
  height: 1.5rem;
  width: 1.5rem;
  margin: 0;
}

.c3 > svg {
  position: absolute;
  z-index: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  height: 100%;
  width: 100%;
  pointer-events: none;
  fill: #CCC;
}

.c3 > svg[name='checked'] {
  display: none;
  z-index: 1;
  fill: #4E3FCE;
}

.c3 > input:checked ~ svg[name='checked'] {
  display: block;
}

.c1 {
  cursor: pointer;
}

.c1:hover input:not(:disabled) ~ svg[name='unchecked'] {
  fill: currentColor;
}

<label
  class="c0 c1"
  color="primary"
  display="flex"
  opacity="1"
>
  <div
    class="c2 c3"
  >
    <input
      color="primary"
      theme="[object Object]"
      type="radio"
    />
    <svg
      class="c4"
      color="inherit"
      fill="currentcolor"
      height="24"
      name="checked"
      size="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
    <svg
      class="c5"
      color="inherit"
      fill="currentcolor"
      height="24"
      name="unchecked"
      size="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </div>
  <div
    class="c6"
    color="text"
    font-size="1"
    font-weight="3"
  />
</label>
`;

exports[`Radio component sanity matches themed snapshot 1`] = `
.c4 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: inherit;
}

.c5 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: inherit;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #4E3FCE;
  opacity: 1;
}

.c2 {
  box-sizing: border-box;
}

.c6 {
  box-sizing: border-box;
  color: #3F3D4B;
  font-size: 14px;
  font-weight: 600;
  line-height: 1;
  margin-left: 4px;
  margin-right: 8px;
}

.c3 {
  position: relative;
  display: inline-block;
}

.c3 > input {
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
  z-index: 1;
  display: block;
  height: 1.5rem;
  width: 1.5rem;
  margin: 0;
}

.c3 > svg {
  position: absolute;
  z-index: 0;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  height: 100%;
  width: 100%;
  pointer-events: none;
  fill: #CCC;
}

.c3 > svg[name='checked'] {
  display: none;
  z-index: 1;
  fill: #4E3FCE;
}

.c3 > input:checked ~ svg[name='checked'] {
  display: block;
}

.c1 {
  cursor: pointer;
}

.c1:hover input:not(:disabled) ~ svg[name='unchecked'] {
  fill: currentColor;
}

<label
  class="c0 c1"
  color="primary"
  display="flex"
  opacity="1"
>
  <div
    class="c2 c3"
  >
    <input
      color="primary"
      theme="[object Object]"
      type="radio"
    />
    <svg
      class="c4"
      color="inherit"
      fill="currentcolor"
      height="24"
      name="checked"
      size="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
    <svg
      class="c5"
      color="inherit"
      fill="currentcolor"
      height="24"
      name="unchecked"
      size="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </div>
  <div
    class="c6"
    color="text"
    font-size="1"
    font-weight="3"
  />
</label>
`;
